src/util/userUrlHandling.ts

```diff
@@ -54,9 +56,23 @@ export const getFileIdFromUrl = (
     url: string,
     idParam?: string | string[] | null
 ) => {
-    // currently only id we're extracting from the url, but possible we'll need more
     if (isGoogleDriveUrl(url)) {
         return getGoogleDriveFileId(url, idParam);
+    } else {
+        // ex) "mysite.com/myTraj.simularium?dl=0" -> "myTraj.simularium"
+        const urlSplit = url.split("/");
+        return urlSplit[urlSplit.length - 1].split("?")[0];
+    }
+};
+
+export const getRedirectUrl = (url: string, fileName: string | undefined) => {
+    if (url && fileName && USER_TRAJ_REDIRECTS.includes(url)) {
+        // ex) simularium.allencell.org/viewer?trajFileName=endocytosis.simularium
+        return `${location.origin}${
+            location.pathname
+        }?trajFileName=${fileName}`;
```

Comment:
funny autoformatting 😄   i guess you can't win em all

Reply:
This is why I tried the `+` and `${}` combo in the first place, but it didn't make it look any better hahaha

Reply:
small thing, but you could use location.href instead of location.origin + location.pathname

Reply:
Ah, I used origin + pathname instead of href because href includes the `?trajUrl=blahblah` which I want to drop

Reply:
oh right!